Lesson1 task1

SevryuginMihail opened the pull request:
мой первый pull request
все ли я сделал правильно?

misa130 committed:
добавил класс Human в правильную наконец ветку

misa130 committed:
добавил класс DtoHuman

misa130 committed:
начал делать класс Convertor<T,V> и поменял аккаунт

misa130 committed:
обновил метод convert<T,V> класса Convertor<T,V>

misa130 committed:
убраны попытки в рекурсию, добавлен класс BD, добавлен параметризированный интерфейс

misa130 committed:
обновил все классы и интерфейс

misa130 committed:
добавил вывод объектов в консоль в методе getAllEntity()

DenisChuprynin commented on .gitignore:
```diff
@@ -24,3 +24,6 @@ hs_err_pid*
 
 /.idea/
 /epam/epam.iml
+/epam/target/
```
А почему все внутри папки epam? она не нужна

DenisChuprynin commented on .gitignore:
```diff
@@ -24,3 +24,6 @@ hs_err_pid*
 /.idea/
 /epam/epam.iml
+/epam/target/
+/epam/src/main/resources/task1
```
для чего добавил ресурсы в игнор?

DenisChuprynin commented on .gitignore:
```diff
@@ -24,3 +24,6 @@ hs_err_pid*
 /epam/epam.iml
+/epam/target/
+/epam/src/main/resources/task1
+/epam/src/test/
```
почему тесты в игноре?

DenisChuprynin commented on epam/src/main/java/Sevryugin/Convertor.java:
```diff
@@ -0,0 +1,42 @@
+import lombok.Data;
+
+@Data
+public class Convertor {
```
Для начала просто ConvertEr

DenisChuprynin commented on epam/src/main/java/Sevryugin/Convertor.java:
```diff
@@ -0,0 +1,42 @@
+
+@Data
+public class Convertor {
+    public Human convert(DtoHuman dtoHuman) {
```
Куча методов conver, какой вызывать и для чего догадайся сам? Давай разнесем этот конвертер на два отдельных? Один будет работать из дто в сущность, другой обратно

DenisChuprynin commented on epam/src/main/java/Sevryugin/Convertor.java:
```diff
@@ -0,0 +1,42 @@
+    public Human[] convertMas(DtoHuman[] dtoHumans) {
+        Human humans[] = new Human[dtoHumans.length];
+        for (int i = 0; i < dtoHumans.length; i++) {
+            humans[i] = new Human(dtoHumans[i].getId(),dtoHumans[i].getName(), dtoHumans[i].getBirthDate().getYear(), dtoHumans[i].getBirthDate().getMonth(),
```
У тебя же уже есть метод который конвертирует одну сущность, давай будем использовать его и избавляться от дублирования кода?

DenisChuprynin commented on epam/src/main/java/Sevryugin/Convertor.java:
```diff
@@ -0,0 +1,42 @@
+    public DtoHuman[] convertMas(Human[] humans) {
+        DtoHuman dtoHumans[] = new DtoHuman[humans.length];
+        for (int i = 0; i < humans.length; i++) {
+            dtoHumans[i] = new DtoHuman(humans[i].getId(),humans[i].getName(), humans[i].getBirthDate().getYear(), humans[i].getBirthDate().getMonth(),
```
тоже самое про дублирование кода

DenisChuprynin commented on epam/src/main/java/Sevryugin/DtoHuman.java:
```diff
@@ -0,0 +1,69 @@
+    private DtoAddress dtoAddress;
+
+    public DtoHuman() {
+        this.id = idConst;// или ++idConst?
```
Не нужно задавать дефолтные значения, в требованиях такого не было

DenisChuprynin commented on epam/src/main/java/Sevryugin/DtoHuman.java:
```diff
@@ -0,0 +1,69 @@
+        this.dtoAddress = new DtoAddress(houseNumber, street, apartmentNumber, town, country);
+    }
+
+    public DtoHuman createRandomDtoHuman() {
```
Этому тут не место, генерация должна быть где-то в другом месте, а конкретно в методе main

DenisChuprynin commented on epam/src/main/java/Sevryugin/DtoHuman.java:
```diff
@@ -0,0 +1,69 @@
+        private String town;
+        private String country;
+
+        DtoAddress() {
```
тоже самое про дефолтные значения

DenisChuprynin commented on epam/src/main/java/Sevryugin/Human.java:
```diff
@@ -0,0 +1,69 @@
+    private String town;
+    private String country;
+
+    Address() {
```
опять про дефолтные значения

DenisChuprynin commented on epam/src/main/java/Sevryugin/Human.java:
```diff
@@ -0,0 +1,69 @@
+        this.address = new Address(houseNumber, street, apartmentNumber, town, country);
+    }
+
+    public Human createRandomHuman() {
```
этому место в репозитории при генерации рандомных сущностей
класс не должен знать как себя генерировать

misa130 committed:
.gitignore:
убрал resourses из игнора
убрал tests из игнора
Convertor:
убрал методы, которые конвертировали массивы сущностей
переназвал методы конвертации одиночных сущностей
Human и DtoHuman:
убрал поле дефолтных значений
убрал метод генерации полей объекта
Service:
добавил метод генерации одного объекта dtoHuman

misa130 committed:
.gitignore:
убрал resourses из игнора
убрал tests из игнора
Convertor:
убрал методы, которые конвертировали массивы сущностей
переназвал методы конвертации одиночных сущностей
Human и DtoHuman:
убрал поле дефолтных значений
убрал метод генерации полей объекта
Service:
добавил метод генерации одного объекта dtoHuman
